packages/suite/src/utils/wallet/transactionUtils.ts

```diff
@@ -367,3 +368,124 @@ export const enhanceTransaction = (
+        t.targets.forEach(target => {
+            target.addresses?.forEach(address => {
+                if (!addresses[address]) {
+                    addresses[address] = [];
+                }
+
+                addresses[address].push(t.txid);
+            });
+        });
+    });
+
+    return addresses;
+};
+
+const groupTransactionsByLabel = (accountMetadata: AccountMetadata) => {
+    const labels: { [label: string]: string[] } = {};
+    const { outputLabels } = accountMetadata;
+
+    Object.keys(outputLabels).forEach(txid => {
+        Object.values(outputLabels[txid]).forEach(label => {
+            if (!labels[label]) {
+                labels[label] = [];
+            }
+
+            labels[label].push(txid);
+        });
+    });
+
+    return labels;
+};
+
+const groupAddressesByLabel = (accountMetadata: AccountMetadata) => {
+    const labels: { [label: string]: string[] } = {};
+    const { addressLabels } = accountMetadata;
+
+    Object.keys(addressLabels).forEach(address => {
+        const label = addressLabels[address];
+
+        if (!labels[label]) {
+            labels[label] = [];
+        }
+
+        labels[label].push(address);
+    });
+
+    return labels;
+};
+
+export const searchTransactions = (
+    transactions: WalletAccountTransaction[],
+    accountMetadata: AccountMetadata,
+    search: string,
+) => {
+    if (search === '') {
+        return transactions;
+    }
+
+    // If it's an amount search (starting with < or > operator)
+    const amountOperator = ['<', '>'].find(k => search.startsWith(k));
+    if (amountOperator) {
+        const amount = new BigNumber(search.replace(amountOperator, ''));
+        switch (amountOperator) {
+            case '<':
+                return transactions.filter(t =>
+                    new BigNumber(t.amount).isLessThanOrEqualTo(amount),
+                );
+            case '>':
+                return transactions.filter(t =>
+                    new BigNumber(t.amount).isGreaterThanOrEqualTo(amount),
+                );
+            // no default
+        }
+    }
+
+    const txsToSearch: string[] = [];
+
+    // Find by output label
+    const txsForOutputLabels = groupTransactionsByLabel(accountMetadata);
+    const foundTxsForOutputLabel = Object.keys(txsForOutputLabels).flatMap(label => {
+        if (label.toLowerCase().includes(search.toLowerCase())) {
+            return txsForOutputLabels[label];
+        }
+
+        return [];
+    });
+    txsToSearch.push(...foundTxsForOutputLabel);
+
+    // Find by address label
+    const addressesForLabel = groupAddressesByLabel(accountMetadata);
+    const foundAddressesForLabel = Object.keys(addressesForLabel).flatMap(label => {
+        if (label.toLowerCase().includes(search.toLowerCase())) {
+            return addressesForLabel[label];
+        }
+
+        return [];
+    });
+
+    // Find by address
+    const txsForAddresses = groupTransactionIdsByAddress(transactions);
```

Comment:
searching by address does not work

example: open tx detail of first tx, go to input/outputs tab, search for one of the address

Reply:
just found the reason, `groupTransactionIdsByAddress` should also look up to [tx.details](https://github.com/trezor/trezor-suite/blob/develop/packages/blockchain-link/src/types/common.ts#L77 )vin/vout.address

Reply:
Addressed in 87a7a03